tree/dataframe/inc/ROOT/RDFHelpers.hxx

```diff
@@ -227,11 +222,10 @@ RResultMap<T> VariationsFor(RResultPtr<T> resPtr)
    // RJittedFilters
    resPtr.fLoopManager->Jit();
 
-   std::shared_ptr<RDFInternal::RActionBase> action = resPtr.fActionPtr;
+   std::shared_ptr<RDFInternal::RActionBase> nominalAction = resPtr.fActionPtr;
 
    // clone the result once for each variation
-   std::vector<std::string> variations = action->GetVariations();
-   variations.insert(variations.begin(), "nominal");
+   std::vector<std::string> variations = nominalAction->GetVariations();
```

Comment:
Is it worth guarding the whole "handle variations" section by an "if we have variations" block?

Reply:
we could avoid creating the `RVariedAction` node now that it only handles varied results and not the nominal one (if there are no variations for a result, the `RVariedAction` is now useless). however that would require teaching `RResultMap` how to deal with a null `fVariedAction`.

I would leave this for another PR if that's ok